packages/frontend-app-api/src/wiring/createApp.tsx

```diff
@@ -351,7 +409,7 @@ function createLegacyAppContext(plugins: BackstagePlugin[]): AppContext {
     },
 
     getComponents(): AppComponents {
-      return defaultComponents;
+      return overriddenComponents;
```

Comment:
Still adding backwards compatibility with the legacy app context so plugins that use `useApp` hooks continue to work. But I would like to confirm if we want to get rid of the `AppContext`, it seems that we are gradually breaking it into separate contexts like we did for theme, routes and now components. One idea that came to mind would be to make the legacy application use the new context behind the scenes. Any thoughts?

Reply:
So when the `useApp` is called by a legacy code, it will get the components from the new `ComponentsContext` instead of getting it from the AppContext in both legacy and new apps.

Reply:
https://github.com/backstage/backstage/pull/21592